### PR TITLE
[DO NOT MERGE] Backport some fixes from master to ozone-1.4

### PR DESCRIPTION
**Pull request by xichen01**
## What changes were proposed in this pull request?

Backport the following fixes from master to ozone-1.4 for 1.4.1 (Do not include Etag-related PRs, these will be backported in separate PRs):

  HDDS-10680. Duplicate delete key blocks sent to SCM (#6513)
  HDDS-10692. ozone s3 getsecret prints some internal details (#6531)
  HDDS-10630. Add missing parent directories deleted between initiate and complete MPU (#6496)
  HDDS-10677. Bump plotly.js to 2.25.2 (#5915)
  HDDS-10676. Bump browserify-sign to 4.2.3 (#6509)
  HDDS-10674. Bump ip to 1.1.9 (#6243)
  HDDS-10673. Bump es5-ext to 0.10.64 (#6284)
  HDDS-10672. Bump express to 4.19.2 (#6442)
  HDDS-10669. Bump axios to 0.28.0 (#6242)
  HDDS-9200. [Snapshot] Added logs and metrics for snapshot purge and set property APIs (#6453)
  HDDS-10590. [Snapshot] Synchronized snapshot purge, set snapshot property and SstFilteringService (#6456)
  HDDS-9426. Calculate Exclusive size for deep cleaned snapshot's deleted directories (#6099)
  HDDS-10477. Make Rocksdb tools native lib compatible with all chipset with the same arch (#6341)
  HDDS-10552. Downgrade Surefire to 3.0.0-M4 (#6406)
  HDDS-9802. Tool to fix corrupted snapshot chain (#6386)
  HDDS-10496. Fetch dependencies for cache using actual build (#6359)

## How was this patch tested?
https://github.com/xichen01/ozone/actions/runs/8732721413

### CODE REVIEW

**Comment by adoroszlai** [comment]
Thanks @xichen01 for working on the backport.  There is already a PR for EC-related fixes: #6551.  Let's rebase this one when that's merged.

**Comment by xichen01** [comment]
https://issues.apache.org/jira/browse/HDDS-10681
https://issues.apache.org/jira/browse/HDDS-10682
https://issues.apache.org/jira/browse/HDDS-10704

Remove above PRs which have been merged in EC-related fixes: https://github.com/apache/ozone/pull/6551

**Comment by xichen01** [comment]
> Thanks @xichen01 for working on the backport. There is already a PR for EC-related fixes: #6551. Let's rebase this one when that's merged.

Done

**Comment by adoroszlai** [comment]
Thanks @xichen01 for updating the PR, my part LGTM.  (There is some minor difference in HDDS-10552 due to HDDS-10013 not being backported, but I think that's OK.)

Please remember to push from CLI instead of merging in GitHub UI.  Also, please do not cherry-pick each commit to `ozone-1.4` from your own branch.  We should preserve the commit hashes from this PR:

```
git checkout ozone-1.4
git pull
git merge --ff-only ozone-1.4-backport
git push
```

**Comment by adoroszlai** [comment]
Let's merge #6557 first, which is already based on top of `ozone-1.4` while this PR is not.  Rebase this one only afterwards.

**Review by hemantk-12 — COMMENTED** [review]
Thanks @xichen01 for the backporting the changes.

I reviewed snapshot related following changes and left one comment. Otherwise it looks good to me.

1. HDDS-9200. [Snapshot] Added logs and metrics for snapshot purge and set property APIs (#6453)
1. HDDS-10590. [Snapshot] Synchronized snapshot purge, set snapshot property and SstFilteringService (#6456)
1. HDDS-9426. Calculate Exclusive size for deep cleaned snapshot's deleted directories (#6099)
1. HDDS-10477. Make Rocksdb tools native lib compatible with all chipset with the same arch (#6341)
1. HDDS-9802. Tool to fix corrupted snapshot chain (#6386)

**Review by ivandika3 — COMMENTED** [review]
Thank you @xichen01 for the backport. 

Could you also help to backport the recent fix HDDS-10717?

**Comment by xichen01** [comment]
> Thank you @xichen01 for the backport.
> 
> Could you also help to backport the recent fix [HDDS-10717](https://issues.apache.org/jira/browse/HDDS-10717)?

Done.

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/SnapshotInfo.java [hunk 1]
```diff
@@ -123,6 +123,7 @@ public static SnapshotStatus valueOf(SnapshotStatusProto status) {
   private long referencedReplicatedSize;
   private long exclusiveSize;
   private long exclusiveReplicatedSize;
+  private boolean deepCleanedDeletedDir;
```
[the remark this answers is not in the document]
Comment (xichen01) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/SnapshotInfo.java:126]: Add to [HDDS-9426.](https://github.com/apache/ozone/pull/6553/commits/2f002cdf9c6cd182f391405ecdc4eb9b72934783)

**Comment by SaketaChalamchala** [comment]
Thanks @xichen01 . My part looks good.

**Comment by adoroszlai** [comment]
Thanks @xichen01 for completing this backport.

**Pull request merged**